src/backend/marsha/core/api/video.py

```diff
@@ -465,6 +475,105 @@ def update_live_state(
 
         return Response({"success": True})
 
+    @action(
+        methods=["get"],
+        detail=True,
+        url_path="pairing-secret",
+        permission_classes=[
+            permissions.IsTokenResourceRouteObject & permissions.IsTokenInstructor
+            | permissions.IsTokenResourceRouteObject & permissions.IsTokenAdmin
+        ],
+    )
+    # pylint: disable=unused-argument
+    def pairing_secret(
+        self,
+        request,
+        pk=None,
+    ):
+        """Generate a secret for pairing an external device to a live stream.
+
+        Parameters
+        ----------
+        request : Type[django.http.request.HttpRequest]
+            The request on the API endpoint
+        pk: string
+            The primary key of the video
+
+        Returns
+        -------
+        Type[rest_framework.response.Response]
+            HttpResponse with the generated secret.
+        """
+        video = self.get_object()
+        try:
+            video.live_info.get("jitsi").get("domain")
+        except AttributeError:
+            return Response(
+                {"detail": "Matching video is not a Jitsi Live."}, status=410
+            )
+
+        try:
+            live_pairing = LivePairing.objects.get(video=video)
+        except LivePairing.DoesNotExist:
+            live_pairing = LivePairing(video=video)
+        live_pairing.generate_secret()
+        live_pairing.save()
+
+        serializer = serializers.LivePairingSerializer(instance=live_pairing)
+        return Response(serializer.data)
+
+
+class PairingChallengeThrottle(AnonRateThrottle):
+    """Throttling for pairing challenge requests."""
+
+    def get_ident(self, request):
+        """Identify requests by box_id instead of IP"""
+        return request.data.get("box_id")
+
+
+@api_view(["POST"])
+@throttle_classes([PairingChallengeThrottle])
+def pairing_challenge(request):
+    """View handling pairing challenge request to stream from an external device.
+
+    Parameters
+    ----------
+    request : Type[django.http.request.HttpRequest]
+        The request on the API endpoint, it should contain a payload with the following fields:
+            - box_id: uuid from external device.
+            - secret: generated pairing secret.
+
+    Returns
+    -------
+    Type[rest_framework.response.Response]
+        HttpResponse containing live video credentials.
+
+    """
+
+    serializer = serializers.PairingChallengeSerializer(data=request.data)
+    if not serializer.is_valid():
+        return Response(serializer.errors, status=400)
+
+    try:
+        live_pairing = LivePairing.objects.get(secret=request.data.get("secret"))
+    except LivePairing.DoesNotExist:
+        return Response({"detail": "Secret not found."}, status=404)
+
+    if live_pairing.expired:
+        live_pairing.delete()
+        return Response({"detail": "Secret expired."}, status=410)
+
+    try:
+        jitsi_domain = live_pairing.video.live_info.get("jitsi").get("domain")
```

Comment:
This is wrong. The `jitsi` domain is never saved in the `video` model. It is computed in the `serializer` https://github.com/openfun/marsha/blob/69431e5ae0663c1a779556a7575352d803688d26/src/backend/marsha/core/serializers/video.py#L669

You must use `settings.JITSI_DOMAIN` here.